### DISCUSSION TITLE
Ads on MDN

### DISCUSSION THREAD

**Discussion by SphinxKnight** [mdn/mdn-community#335] [MDN Plus] [answered]
https://developer.mozilla.org/en-US/advertising
https://developer.mozilla.org/en-US/advertising/with_us
https://github.com/mdn/yari/pull/8213

Just adding this as pointer now, as I thought it would have been largely communicated by now. Also creating a thread for the "occasion"

**Comment by schalkneethling** [comment]
Hey @SphinxKnight, thank you for stating the thread. Do you or anyone else from the community have any questions about this topic?

**Comment by SphinxKnight** [comment]
Disclaimer: it will soon be 10 years I contribute to MDN, esp. localizing its content in French. What follows is my opinion only and does not constitute a statement for the MDN l10n-fr community.

This content is also hosted on https://gist.github.com/SphinxKnight/f8d86b6bdc2910719ba3b1476ac09910

## Ethics & Transparency

### Transparency

There has been 0 announcements to the public nor to the community. Here is the 8th principle of the Mozilla Manifesto for anyone interested to read: "Transparent community-based processes promote participation, accountability and trust." 

I have since seen several messages from my end (without actively looking for them) "was there a blog post or an annoucement?".

https://github.com/mdn/yari/pull/8213 has no rationale nor explanation. In my mind "open source" is more than just "share source code on GitHub and deal with it". It would be nice for yari to be exemplary in terms of intent/transparency in order to build trust.

A few weeks ago (and before ads inclusion), a roadmap with OWD was advertised: no mention of ads in [the blog post](https://blog.mozilla.org/en/mozilla/mdn-web-documentation-collaboration/) nor [in the tracking GH view](https://github.com/orgs/mdn/projects/26/views/6). On the other hand, this roadmap is advertised right from https://developer.mozilla.org/en-US/advertising. Is this roadmap a means to an end (generating revenue) or the contrary?

To balance this, I strongly hope there will be a report at the end of the six weeks of this "experiment" (29th of March) and that any continuation will not be made in silence by default.

### Ethics & other commons

On https://www.kevel.com/, we can read *on the front page* "Bypass ad blockers with ad server APIs" which, to me, does not align with "user respect".

On the other hand, neither Wikipedia nor OpenStreetMap have ads while those noble projects:
- Share some foundational ideas with MDN (open knowledge/contribution, etc.).
- have apparently less financial resources than MoCo

Mozilla also has another "common" project that I know of: Common Voice. As far as I understand, this project is lead by Mozilla Foundation (whereas MDN is lead by Mozilla Corporation): would/should/could MDN switch to MoFo governance?

## Accountability & distribution

Contrary to previous MDN Plus features, this "feature" affects every reader on MDN. For [other features](https://developer.mozilla.org/en-US/plus#features), it was sensible for Mozilla to collect the corresponding revenue as they were the sole creators and maintainers of MDN Plus. Since ads affect every single page (whether English or localized, whether written by a MoCo employee, an OWD member, or a volunteer): what is the plan to distribute the revenue generated by the ads among the diverse community writing them?

For instance, [in the past, 40% of MDN traffic](https://github.com/mdn/translated-content/discussions/1667) was generated from non-English content. Does that mean that Mozilla will "invest" 40% (minus a margin for hosting/managing the platform) of ads revenue to maintaining localization tooling/communities for MDN?

A similar question could be asked about the OWD "share".

### Efficiency

I have no statistics regarding the developer population, but I would tend to think that, as most of them are tech-savvy, they have an ad blocker installed (this question could have been asked beforehand on Matrix/Discord). If I understand it correctly: blocking the ads would block the revenue. 

Moreover:
- Mozilla recommends using µBlock Origin and this is the most used extension for Firefox: https://addons.mozilla.org/en-US/firefox/search/?promoted=recommended&sort=users&type=extension (see also https://blog.mozilla.org/en/uncategorized/online-advertising-strategies/)
- https://github.com/easylist/easylist/commit/f02f2264cdc4122c1af971519cd640931d1121ef MDN ads were added to the easylist less than 36h after https://github.com/mdn/yari/pull/8213 was merged, meaning most of µBlock origin users with default settings (myself included) will never see an ad. This heavily "reduces" the value one can get from an MDN Plus subscription if the intent is not to receive ads.

In regards to all of this, I would not bet that much on ads and do not quite understand Mozilla strategy here.
I hope MDN Plus is sustainable at least for the resources Mozilla put in its development.

## Prospective

I can't help but ask those bleak questions.

- What if this does not bring "enough" (to be defined) revenue? What would Mozilla do?
  - Corollary: the recent addition of ads could indicate the existing revenue flow from MDN Plus was previously not sufficient regarding MoCo objectives.
- Does Mozilla intend to let OWD "run" MDN in the near future?
  - Does Mozilla intend to "donate" the project to another foundation?
  - Aside: it seems like the Open Collective "economic model" is pretty helpful for commons, maybe the OWD team (or OC) would have some interesting feedback about sustaining a common like MDN and its platform.
- What does Mozilla think of alternative/forks for providing content under CC-BY-SA 2.5?

## References

- [https://developer.mozilla.org/en-US/advertising/with_us](https://web.archive.org/web/20230227232832/https://developer.mozilla.org/en-US/advertising/with_us)
- [https://developer.mozilla.org/en-US/advertising](https://web.archive.org/web/20230227232832/https://developer.mozilla.org/en-US/advertising)
- [https://www.kevel.com/](https://web.archive.org/web/20230301150948/https://www.kevel.com/)
- https://blog.mozilla.org/en/mozilla/mdn-web-documentation-collaboration/

**Comment by schalkneethling** [comment]
> **NOTE:** The following represents the collective feedback of the MDN team at Mozilla, not any individual.

First of all, thank you for taking the time to provide feedback. We also want to thank you, and our wider community, for all your support and contributions.

> There has been 0 announcements to the public nor to the community. Here is the 8th principle of the Mozilla Manifesto for anyone interested to read: "Transparent community-based processes promote participation, accountability and trust."

In general, Mozilla does not do outreach for experiments on our products. This is partially related to limited resources and the high number of experiments across Mozilla, and partially because we do not want to focus people’s attention on the experiment but instead see how it performs on its own. This is why your feedback is so valuable. 

We added a page on MDN Web Docs, linked from the display ad, where we explained the why and how of advertising on MDN. We will review the experiment results at the end of March, and based on this, we will share the next steps. At that point, if there is a launch plan, it will involve widely communicating the why, how, and other relevant details.

> On kevel.com, we can read on the front page "Bypass ad blockers with ad server APIs" which, to me, does not align with "user respect".

We cannot speak directly to Kevel’s marketing copy decisions. What we can do is explain how we implemented ads. We are partnering with Kevel because they allow us to make privacy-compliant ads without sharing user data.

The ads are served based on the page’s context and not through user tracking. We use country data, which is done on our CDN and not shared with any third party. We use our own ads server, and even that one does not get the IP of the user, only the country code. To get a little more technical 😊, we use an IP range and not the user’s specific IP address. This was custom work done by engineering to protect user privacy further.

Also, one thing to note is that, as you later mention, if you have uBlock installed with EasyList, ads on MDN Web Docs are indeed blocked. 

> Mozilla also has another "common" project that I know of: Common Voice. As far as I understand, this project is lead by Mozilla Foundation (whereas MDN is lead by Mozilla Corporation): would/should/could MDN switch to MoFo governance?

MDN Web Docs is part of the Mozilla corporation and incurs costs that Mozilla has covered for the past 17 years. Mozilla recognizes the value of MDN Web Docs for the wider web developer community. We all have MDN’s future at heart and want to ensure that the project will continue to serve web developers worldwide for many years to come. 

Mozilla, as a company, has a long-standing goal of diversifying its revenue streams. This will involve experimenting in different areas, and success in that goal should ultimately strengthen Mozilla and its mission. 

A delicate balance must be struck here, so we rely on our community of readers and contributors to help guide us along this path. As with most things, we will not always agree on everything, but we want to ensure the balance does not tip too much in either direction.

> Does that mean that Mozilla will "invest" 40% (minus a margin for hosting/managing the platform) of ads revenue to maintaining localization tooling/communities for MDN? A similar question could be asked about the OWD "share".

Our focus is to continually make MDN better - that is why we have a fully scoped roadmap with pure MDN product improvements out there that we are committed to delivering by the end of the year. Ideally, this impacts our investment in our relationship with partners and the community. There are active conversations about improving our tooling and ways in which we support our partners and community. 

> In regards to all of this, I would not bet that much on ads and do not quite understand Mozilla strategy here. I hope MDN Plus is sustainable at least for the resources Mozilla put in its development.

We believe that advertising can be effective and is a major part of how the internet is financed today. We do our best to display ads that we believe are relevant in context and come from companies that align with Mozilla’s vision and values. We do not track or share personal information, and we do believe this could be a great option for sustaining MDN moving forward without sacrificing user experience on the site.

> Does Mozilla intend to let OWD "run" MDN in the near future?

MDN Web Docs is a collective effort, and we believe that will always be true. OWD is one of our closest partners, together with the Product Advisory Board. There are also many individual contributors (like yourself) who are invaluable to the project. 

Mozilla has been continuously investing in MDN for the last 17 years, we are committed to the project, and to answer your specific question, we have no intention to let OWD or anyone else ‘run’ MDN in the near future. 

> Does Mozilla intend to "donate" the project to another foundation?

As mentioned earlier, Mozilla Corporation plans to continue to invest in and host MDN. 

> Aside: it seems like the Open Collective "economic model" is pretty helpful for commons, maybe the OWD team (or OC) would have some interesting feedback about sustaining a common like MDN and its platform.

As mentioned before, not something we are actively pursuing at the moment, but we thank you for the suggestion.

> What does Mozilla think of alternative/forks for providing content under CC-BY-SA 2.5?

As an open-source project, and with the licenses chosen, we are aware that anyone is able to fork the MDN Web Docs project and start it under a different name, brand, business, and contribution model. We hope the community we have built over the past 17+ years will stick around and help ensure that the current MDN Web Docs keep getting better for everyone.

Thank you again for your thoughtful feedback.

**Comment by SphinxKnight** [comment]
It looks like this is not an experiment anymore. 
Communication and transparency have not been a forte in this case.
![image](https://user-images.githubusercontent.com/2413436/231369814-e0c0d8ba-5c62-40c9-aab2-8ca896a48d4b.png)

**Reply by schalkneethling** [reply, marked as answer]
Hey @SphinxKnight, here is a summary of what we talked about in the community call. Thanks to those who could attend.

Apologies that it took so long to post this. There is a _lot_ going on at the moment.

Hermina talked about the ads experiment. The ads experiment was a success, and ads will continue on MDN. Florian (Engineering team) demonstrated a new code playground that is being worked on. Dipika (Content team) provided some highlights of the work that the content team has been doing, such as the improvements to the RegExp pages, new guidelines for the “See also” section of our pages, and the addition of new examples to the CSS color function page.

We ended by announcing that the MDN Web Docs blog will be launched on 1 May this year. The blog will be used to publish articles that demonstrate how multiple open web technologies are used to build websites and apps. Through the blog, we will also provide regular updates on what the MDN Web Docs team has been working on, including updates from the engineering team., The Community Spotlight contributors will now be featured here. The blog will also be used to feature sponsored content. We have strict guidelines around sponsored content. It will be clearly identified as such, the content provided by the partner has to be valuable to our readers by itself, but also, the service promoted needs to make sense for our readers.

We also announced that we are working on a project called the MDN Web Docs sustainability initiative. This initiative is our way to support and give back to the maintainers and open-source projects that MDN Web Docs relies on. We will also support not-for-profit organizations focused on empowering underrepresented people in tech, such as Women Who Code, and educational projects like freeCodeCamp. What this support will look like is not decided as we need to speak with these people and organizations to understand how this initiative can be the most effective for them.

**Reply by schalkneethling** [reply]
There was a question on Discord:

> Regarding the blog, will its URL live under [developer.mozilla.org](http://developer.mozilla.org/) and/or the content live in the mdn/content repo? why was https://hacks.mozilla.org/ not an option?

Yes, it will be at https://developer.mozilla.org/blog but the content will not be part of mdn/content. The MDN team does not own Hacks, and it is used for other updates that are unrelated to MDN. We wanted to create a space that is specifically focused on MDN and what we, the community, and our partners do. As mentioned in the update, we also want to use the blog to provide other value-added content through sponsored content, which would not be possible on Hacks.
